### PR TITLE
Fix generation of wrong field reference in scale. Also add a test for it.

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: src/compile/scale.ts [hunk 1]
```diff
@@ -122,7 +123,7 @@ function parseBinColorLegendLabel(model: Model, fieldDef: FieldDef) {
     type: 'ordinal',
     domain: {
       data: model.dataTable(),
-      field: model.field(COLOR,  {prefn: 'rank_'}),
+      field: model.field(COLOR, (fieldDef.bin || fieldDef.timeUnit) ? {} : { prefn: 'rank_' }),
```
Comment (kanitw) [inline src/compile/scale.ts]: @domoritz   Now I know when this got wrong.  

On this line, `fieldDef.bin` is always `true`.  (See how `parseBinColorLegendLabel` gets called.)  So I asked you to remove `fieldDef.bin`, but I guess you put in the wrong condition.


I think the right fix here is `model.field(COLOR, {})`

File: test/compile/scale.test.ts [hunk 2]
```diff
@@ -349,6 +349,11 @@ describe('Scale', function() {
       assert.deepEqual(scales[1].range, {
         data: 'source',
         field: 'bin_origin_range',
+        sort: { "field": "bin_origin_start", "op": "min" }
```
Comment (kanitw) [inline test/compile/scale.test.ts]: why do we have to assert the `scales` twice?

Comment (domoritz) [inline test/compile/scale.test.ts]: I meant domain. Ups.

**Comment by domoritz** [comment]
Done

**Pull request merged**